pyiron_contrib/RDM/storagejob.py

```diff
@@ -0,0 +1,272 @@
+        else:
+            del self._stored_files[file]
+
+    def remove_files(self, filenames, dryrun=True, raise_error=True, silent=False):
+        """Remove files in the storage
+
+        Parameters:
+            filenames(str/list of str):     Files to remove from the Storage
+            dryrun(bool):                   If true, only report the files to be removed (default=True)
+            raise_error(bool):              If true, missing files raise a FileNotFoundError (nothing gets removed),
+                                            else remove all files found.
+            silent(bool):                   If true, do not print the status message.
+        """
+        if isinstance(filenames, str):
+            filenames = [filenames]
+
+        files_not_found = [file for file in filenames if file not in self._stored_files]
+        if len(files_not_found) > 0 and raise_error:
+            raise FileNotFoundError(f"Files {files_not_found} not found, abort removing files. You may specify "
+                                    f"raise_error=False to suppress this error.")
+
+        files_to_remove = [file for file in filenames if file in self._stored_files]
+        if dryrun and not silent:
+            print(f"dryrun: to remove {files_to_remove} specify dryrun=False")
+            return
+
+        for file in files_to_remove:
+            self._remove_file(file)
+
+        if not silent:
+            print(f"removed {files_to_remove}")
+        self.run()
+
+    def run_static(self):
+        raise NotImplementedError('There is no static run mode for a Storage job.')
+
+    def collect_output(self):
+        pass
+
+    def append(self, job):
+        pass
+
+    def run_if_interactive(self):
+        self.status.running = True
+        self.to_hdf()
+
+    def interactive_close(self):
+        self.to_hdf()
+
+    def interactive_fetch(self):
+        pass
+
+    def interactive_flush(self, path="generic", include_last_step=True):
+        pass
+
+    def run_if_refresh(self):
+        pass
+
+    def _run_if_busy(self):
+        pass
+
+    def write_input(self):
+        pass
+
+    def from_hdf(self, hdf=None, group_name=None):
+        super().from_hdf(hdf, group_name)
+        self._stored_files.from_hdf(hdf=self._hdf5, group_name=group_name)
+        self._input.from_hdf(hdf=self._hdf5, group_name=group_name)
+        self._storage_type = StorageType(self._input.storage_type, self._input.storage_type_read_only)
+        if self._storage_type.s3 and self.status != "initialized":
+            self._external_storage = FileS3IO(
+                config=self._input.s3.config,
+                path=self.path,
+                bucket_name=self._input.s3.bucket_name
+            )
+
+    def to_hdf(self, hdf=None, group_name=None):
+        super().to_hdf(hdf, group_name)
+        self._stored_files.to_hdf(hdf=self._hdf5, group_name=group_name)
+        self._input.to_hdf(hdf=self._hdf5, group_name=group_name)
```

Comment:
Generally `from_hdf`/to_hdf` expect to write/read themselves into/from `hdf[group_name]` and have that group exclusively for themselves.  In case of the `DataContainer` passing `group_name` overwrites `table_name` as the target HDF group.  Therefore here both `_input` and `_stored_files` try to write to the same group, I think.  That this does not raise an error is a bug.

I would usually write something closer to this
```python
super().to_hdf(hdf=hdf, group_name=group_name)
# from here on self._hdf5 points to hdf[group_name]
self._input.to_hdf(hdf=self._hdf5) # writes to self._hdf5[self._input.table_name]
# OR!
self._input.to_hdf(hdf=hdf, group_name='my_input') # writes to self._hdf5['my_input']

Reply:
Thanks a lot! I did not look up how the group_name is being used in the super call... I will drop the group names in the `DataContainer.to/from_hdf` calls and see if this works out 👍